xarray/backends/common.py

```diff
@@ -184,7 +185,7 @@ def sync(self):
             import dask.array as da
             import dask
             if LooseVersion(dask.__version__) > LooseVersion('0.8.1'):
-                da.store(self.sources, self.targets, lock=GLOBAL_LOCK)
+                da.store(self.sources, self.targets, lock=self.lock)
```

Comment:
FYI, I made this modification to the `ArrayWriter` class to allow us to bypass the use of a lock when writing, since (the way we have implemented it), the writes to the zarr store should all be thread safe and not require any lock. Later on, when we initialize the ZarrStore, we create a writer for it as `ArrayWriter(lock=None)`. @mrocklin, does this seem like the right approach.

Reply:
If you can guarantee that no two tasks will write to the same block in Zarr then yes, I think that it is appropriate to avoid locking.  This is based on old information though.

Reply:
Another observation: this call to `dask.array.store` does not show up on my distributed dashboard task / status pages. The writes are obviously happening, but I can't follow their progress. Since we are going to be using this to move some very large datasets, interactive monitoring via the dashboard is quite important. Do you have any idea why this is not showing up?

Reply:
> If you can guarantee that no two tasks will write to the same block in Zarr then yes, I think that it is appropriate to avoid locking.

As of this PR, we do not allow multiple dask chunks per zarr chunk. That scenario is covered by the test suite. It may change in the future, but that's how it is for now.

Once we cross that bridge, we will also have to deal with the fact that both zarr and dask have their own locking (aka "synchronization" in zarr parlance) enforcement mechanisms. We will presumably have to pick one.

Reply:
> Another observation: this call to dask.array.store does not show up on my distributed dashboard task / status pages. The writes are obviously happening, but I can't follow their progress. Since we are going to be using this to move some very large datasets, interactive monitoring via the dashboard is quite important. Do you have any idea why this is not showing up?

There is no reason that a task run on the distributed system will not show up on the dashboard.  My first guess is that somehow you're using a local scheduler.

Reply:
> As of this PR, we do not allow multiple dask chunks per zarr chunk. That scenario is covered by the test suite. It may change in the future, but that's how it is for now.

> Once we cross that bridge, we will also have to deal with the fact that both zarr and dask have their own locking (aka "synchronization" in zarr parlance) enforcement mechanisms. We will presumably have to pick one.
 
I suspect that we will never change this behavior.  I don't think we should ever have multiple dask chunks write to one zarr chunk.  Any `to_zarr` storage function should rechunk explicitly.

If for some reason we did need to synchronize, Dask provides a distributed locking mechanism that could be keyed by chunk label.

Reply:
> There is no reason that a task run on the distributed system will not show up on the dashboard. My first guess is that somehow you're using a local scheduler.

I was not using a local scheduler. After digging further, I can see the tasks on the distributed dashboard using a regular `zarr.DirectoryStore`, but not when I pass a `gcsfs.mapping.GCSMap` to `to_zarr`. Is there any reasons these two should behave differently?

Reply:
Update: it does eventually show up, it just takes a really long time.

Reply:
Yes, this looks good.